### ISSUE TITLE
varnish_param does not work on Debian Jessie

### ISSUE THREAD

**Issue by cjeanneret** [camptocamp/puppet-varnish#20]
Examples of non-working stuff:
```Puppet
  class {'::varnish':
    listen_port     => 80,
    multi_instances => false,
    secret_file     => '/etc/varnish/secret',
    storage         => 'file,/var/cache/varnish/varnish_storage.bin,98%',
    vcl_content     => file("module/foo"),
  }
```
Expected result: should set Varnish port to 80, set Varnish storage engine to "file" with the other parameter.

Actual result:
- puppet fails to set the port:
```Puppet
Notice: /Stage[main]/Varnish::Config/Varnish_param[listen_port]/ensure: created
Error: /Stage[main]/Varnish::Config/Varnish_param[listen_port]: Could not evaluate: Failed to save Augeas tree to file. See debug logs for details.
Notice: /Stage[main]/Systemd/Exec[systemctl-daemon-reload]: Dependency Varnish_param[listen_port] has failures: true
Warning: /Stage[main]/Systemd/Exec[systemctl-daemon-reload]: Skipping because of failed dependencies
Notice: /Stage[main]/Varnish::Service/Service[varnish]: Dependency Varnish_param[listen_port] has failures: true
Warning: /Stage[main]/Varnish::Service/Service[varnish]: Skipping because of failed dependencies
Notice: /Stage[main]/Profiles_chtopo::App::Varnish_proxy/Service[varnishncsa]: Dependency Varnish_param[listen_port] has failures: true
Warning: /Stage[main]/Profiles_chtopo::App::Varnish_proxy/Service[varnishncsa]: Skipping because of failed dependencies
Notice: Finished catalog run in 13.10 seconds
```

- Puppet fails to set storage engine (nothing in both /etc/default/varnish and /lib/systemd/system/varnish.service)

**Comment by raphink** [comment]
Can you run this with `--debug` to get Augeas debug logs please?

**Comment by cjeanneret** [comment]
```Puppet
LC_ALL=C puppet agent -t --environment cjt_varnish4 --no-report --debug | grep -i varnish_param
Debug: /Stage[main]/Varnish::Config/Varnish_param[listen_address]/notify: subscribes to Exec[systemctl-daemon-reload]
Debug: /Stage[main]/Varnish::Config/Varnish_param[listen_port]/notify: subscribes to Exec[systemctl-daemon-reload]
Debug: /Stage[main]/Varnish::Config/Varnish_param[listen_address]: Nothing to manage: no ensure and the resource doesn't exist
Debug: /Stage[main]/Varnish::Config/Varnish_param[listen_port]: Nothing to manage: no ensure and the resource doesn't exist
```

And nothing regarding the storage… Weird.

**Comment by cjeanneret** [comment]
Some more tests:
* file located in /lib/systemd/system/varnish.service doesn't have any of the settings passed via puppet (especially the "storage", option -s, which is still set by default to malloc,256).
* even after editing this file and reloading varnish service, it's still using the malloc,256 option — seems the /etc/default/varnish is stronger.

Still digging.

**Comment by raphink** [comment]
Fixed in 1.99.19
